### PR TITLE
Rodrigom/benchmarking support fastapi

### PR DESCRIPTION
**Pull request by snova-rodrigom**
- Added compatibility with FastApi in src and streamlit
- Refactored code to handle different API endpoints. Created APIEndpoint abstract class, where SambastudioAPI and FastAPI inherit from
- Tested with CoE, standalone, Turbo and FastAPI endpoints using streaming URIs
- Cleaned and documented code

### CODE REVIEW

File: benchmarking/src/llmperf/sambanova_client.py [hunk 1]
```diff
@@ -21,315 +23,503 @@
...
-        request_config (RequestConfig): contains LLM params
+class SambaStudioAPI(BaseAPIEndpoint):
+    
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Load sambastudio env variables
+        self.base_url = os.environ.get("SAMBASTUDIO_BASE_URL")
+        self.project_id = os.environ.get("SAMBASTUDIO_PROJECT_ID")
+        self.endpoint_id = os.environ.get("SAMBASTUDIO_ENDPOINT_ID")
+        self.api_key = os.environ.get("SAMBASTUDIO_API_KEY")
```
Comment (snova-jorgep) [inline benchmarking/src/llmperf/sambanova_client.py:229]: why is base URI param not used here?

**Review by snova-jorgep — CHANGES_REQUESTED** [review]
minor changes requested and comments

File: benchmarking/README.md [hunk 2]
```diff
@@ -201,14 +214,16 @@ This option allows you to evaluate the performance of the selected LLM on your o
 - The entered path should be an absolute path to your dataset.
   - For example: `/Users/johndoe/Documents/my_dataset.jsonl`
 
-3. Enter a model name
+3. Enter a model name and choose the right API type
```
[the remark this answers is not in the document]
Comment (snova-rodrigom) [reply benchmarking/README.md:217]: yes, those are for different options. First one is for Perf eval on Synthetic data and second Perf eval on Custom data. They're being duplicated so it's easier for users to read in the same section the info they need

File: benchmarking/README.md [hunk 3]
```diff
@@ -49,7 +49,7 @@ This sample is ready-to-use. We provide:
    
 # Before you begin
 
-To perform this setup, you must be a SambaNova customer with a SambaStudio account. You also have to set up your environment before you can run or customize the starter kit. 
+To perform this setup, you must be a SambaNova customer with a SambaStudio account or a FastCOE API key. You also have to set up your environment before you can run or customize the starter kit.
```
[the remark this answers is not in the document]
Comment (snova-rodrigom) [reply benchmarking/README.md:52]: Done, thanks!

File: benchmarking/src/llmperf/sambanova_client.py (hunk 1 above)
Comment (snova-rodrigom) [reply benchmarking/src/llmperf/sambanova_client.py:229]: that change will be included in the PR about Api v2. Please let's follow this on there

**Review by snova-rodrigom — COMMENTED** [review]
Left some comments that should help

**Review by snova-jorgep — APPROVED** [review]

**Comment by snova-jorgep** [comment]
merged!

**Pull request merged**